src/TestingUtils/Microsoft.AspNetCore.Testing/src/HttpClientSlim.cs

```diff
@@ -94,6 +101,33 @@ private static async Task<string> ReadResponse(Stream stream)
             }
         }
 
+        private static async Task<string> RetryRequest(Func<Task<string>> retryBlock)
+        {
+            var retryCount = 1;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                retryCount = 3;
+            }
+
+            for (var retry = 0; retry < retryCount; retry++)
+            {
+                try
+                {
+                    return await retryBlock().ConfigureAwait(false);
+                }
+                catch (Exception)
```

Comment:
Can we retry just when ReadResponse gets no data instead of retrying for any exception?

Reply:
Sure, I'll just retry on `InvalidDataException` which is only thrown if there is no data.